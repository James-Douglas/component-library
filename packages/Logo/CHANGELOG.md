# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.2.3](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-logo@0.2.2...@comparethemarketau/manor-logo@0.2.3) (2020-06-17)

**Note:** Version bump only for package @comparethemarketau/manor-logo





## [0.2.2](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-logo@0.2.1...@comparethemarketau/manor-logo@0.2.2) (2020-06-16)

**Note:** Version bump only for package @comparethemarketau/manor-logo





## [0.2.1](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-logo@0.2.0...@comparethemarketau/manor-logo@0.2.1) (2020-06-15)

**Note:** Version bump only for package @comparethemarketau/manor-logo





# [0.2.0](https://github.com/comparethemarketau/manor-react/compare/@comparethemarketau/manor-logo@0.1.0...@comparethemarketau/manor-logo@0.2.0) (2020-06-12)


### Features

* **toasts:** manorProvider now injects ToastManager ([#333](https://github.com/comparethemarketau/manor-react/issues/333)) ([8621888](https://github.com/comparethemarketau/manor-react/commit/862188867bbc8258b29fa162f46e5ad5b108f778))


### BREAKING CHANGES

* **toasts:** Added ToastNotification component. Refactored toasts such that ManorProvider
injects the ToastManager. addToasts also now expects a <ToastNotification> instead of an object.

* fix(story): fix typo in toast notification story





# 0.1.0 (2020-06-11)


### Features

* **packages:** migrated to monorepo ([920a45e](https://github.com/comparethemarketau/manor-react/commit/920a45ec4b40a19de32f39f29693cbe1b1f314ae))
