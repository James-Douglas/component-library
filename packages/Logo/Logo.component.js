import React from 'react';
import PropTypes from 'prop-types';
import { Picture, picturePropTypes } from '@comparethemarketau/manor-picture';
import StyledLogo from './Logo.styles';

const Logo = ({ size, link, picture }) => {
  const {
    src, srcsets, alt, title, className,
  } = picture;
  return (
    <StyledLogo size={size} id="logo">
      <a href={link}>
        <Picture
          src={src}
          srcsets={srcsets}
          alt={alt}
          title={title}
          className={className}
        />
      </a>
    </StyledLogo>
  );
};

Logo.propTypes = {
  /**
   * Address for the <a> tag wrapping the logo image.
   */
  link: PropTypes.string,
  /**
   * Size of the logo (small = 3.2rem, large = 4.4rem)
   */
  size: PropTypes.oneOf(['small', 'large']),
  picture: PropTypes.shape(picturePropTypes).isRequired,
};

Logo.defaultProps = {
  link: 'https://www.comparethemarket.com.au',
  size: 'large',
};

export default Logo;
