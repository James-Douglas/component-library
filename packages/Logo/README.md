# Logo

## Installation

`yarn add @comparethemarketau/manor-logo`

## Documentation

For technical documentation, see [Storybook](https://services.dev.comparethemarket.cloud/manor/?path=/docs/components-logo--logo).
