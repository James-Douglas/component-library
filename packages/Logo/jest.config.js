const baseConfig = require('../../jest.config.base');

module.exports = {
  ...baseConfig,
  roots: [
    '<rootDir>/packages/Logo',
  ],
  collectCoverageFrom: [
    '*.{js,jsx,ts,tsx}',
  ],
  name: 'Logo',
  displayName: 'Logo',
  rootDir: '../..',
};
