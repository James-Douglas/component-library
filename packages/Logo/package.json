{
  "name": "@comparethemarketau/manor-logo",
  "version": "0.2.0",
  "description": "Manor Logo component",
  "main": "lib/index.cjs.js",
  "author": "Compare The Market",
  "license": "Proprietary",
  "scripts": {
    "build": "rollup -c '../../rollup.config.js'",
    "tdd": "jest --coverage --watchAll",
    "test": "cross-env JEST_SILENT_REPORTER_DOTS=true jest --reporters=jest-silent-reporter --coverage --watchAll",
    "test:ci": "cross-env CI=true jest",
    "prepare": "cp package.json lib/ && cp CHANGELOG.md lib/"
  },
  "publishConfig": {
    "registry": "https://nexus.comparethemarket.com.au/repository/platform-packages/",
    "access": "public",
    "tag": "latest"
  },
  "dependencies": {
    "@comparethemarketau/manor-hooks": "^0.1.0",
    "@comparethemarketau/manor-picture": "^0.1.0",
    "@comparethemarketau/manor-utils": "^0.1.0",
    "prop-types": "^15.7.2"
  },
  "devDependencies": {
    "@comparethemarketau/manor-themes": "^0.1.0"
  },
  "peerDependencies": {
    "react": "^16.11.0",
    "react-dom": "^16.11.0",
    "styled-components": "^5.0.0"
  }
}
