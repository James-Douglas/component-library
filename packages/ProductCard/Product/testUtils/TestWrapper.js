import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
