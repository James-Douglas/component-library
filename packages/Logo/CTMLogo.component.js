import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { ThemeContext } from 'styled-components';
import { ManorContext } from '@comparethemarketau/manor-provider';
import Logo from './Logo.component';
import CTMLogoInline from '../../images/ctm-logo-inline.svg';
import CTMLogoInlineInverted from '../../images/ctm-logo-inline-inverted.svg';
import CTMLogoStacked from '../../images/ctm-logo-stacked.svg';
import CTMLogoStackedInverted from '../../images/ctm-logo-stacked-inverted.svg';

const CTMLogo = ({
  size, inverted, className,
}) => {
  const theme = useContext(ThemeContext);
  const { breakpoint } = useContext(ManorContext);
  let src = CTMLogoStacked;
  if (breakpoint === 'md') {
    src = inverted ? CTMLogoStackedInverted : CTMLogoStacked;
  } else if (breakpoint !== 'xs') {
    src = inverted ? CTMLogoInlineInverted : CTMLogoInline;
  }
  return (
    <Logo
      size={size}
      link="https://www.comparethemarket.com.au"
      picture={{
        src,
        srcsets: [
          {
            srcset: inverted ? CTMLogoStackedInverted : CTMLogoStacked,
            media: `(max-width: ${theme.breakpoints.lg}`,
          },
          {
            srcset: inverted ? CTMLogoInlineInverted : CTMLogoInline,
          },
        ],
        className,
      }}
    />
  );
};

CTMLogo.propTypes = {
  size: PropTypes.oneOf(['small', 'large']),
  inverted: PropTypes.bool,
  className: PropTypes.string,
};

CTMLogo.defaultProps = {
  size: 'large',
  inverted: false,
  className: '',
};

export default CTMLogo;
