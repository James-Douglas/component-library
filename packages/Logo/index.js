import Logo from './Logo.component';
import CTMLogo from './CTMLogo.component';

export { Logo, CTMLogo };
