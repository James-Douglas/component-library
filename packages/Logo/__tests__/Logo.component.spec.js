import React from 'react';
import { ctmTheme } from '@comparethemarketau/manor-themes';
import { screens } from '@comparethemarketau/manor-utils';
import { render } from '../../../testUtils';
import Logo from '../Logo.component';
import CTMLogoInline from '../../../images/ctm-logo-inline.svg';
import CTMLogoStacked from '../../../images/ctm-logo-stacked.svg';

describe('Logo', () => {
  const pictureProps = {
    src: CTMLogoStacked,
    srcsets: [
      {
        srcset: CTMLogoStacked,
        media: `(max-width: ${screens.lg}`,
      },
      {
        srcset: CTMLogoInline,
      },
    ],
  };

  it('renders correctly with minimal props', () => {
    const { container } = render(<Logo picture={pictureProps} />);
    expect(container.innerHTML).toMatchSnapshot();

    const a = container.querySelector('a');
    expect(a).toBeDefined();
    expect(a.getAttribute('href')).toBe('https://www.comparethemarket.com.au');

    const sources = container.querySelectorAll('source');
    expect(sources).toBeDefined();
    expect(sources[0].getAttribute('srcset')).toEqual('ctm-logo-stacked.svg');
    expect(sources[1].getAttribute('srcset')).toEqual('ctm-logo-inline.svg');
  });

  it('renders a small logo', () => {
    const { container } = render(<Logo size="small" picture={pictureProps} />);
    expect(container.querySelector('#logo')).toHaveStyle(`height: ${ctmTheme.logo.heightSmall}`);
  });
});
