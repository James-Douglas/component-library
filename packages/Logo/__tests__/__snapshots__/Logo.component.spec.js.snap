// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly with minimal props 1`] = `"<div id=\\"logo\\" class=\\"sc-AxiKw bRaKDp\\"><a href=\\"https://www.comparethemarket.com.au\\"><picture class=\\"sc-AxjAm kaRYdH\\"><source srcset=\\"ctm-logo-stacked.svg\\" media=\\"(max-width: 1200px\\"><source srcset=\\"ctm-logo-inline.svg\\"><img src=\\"ctm-logo-stacked.svg\\" alt=\\"\\" title=\\"\\" class=\\"sc-AxirZ eWklgb\\"></picture></a></div>"`;
